core/src/main/python/wlsdeploy/tool/deploy/odl_deployer.py

```diff
@@ -0,0 +1,304 @@
+"""
+Copyright (c) 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+"""
+from java.io import File
+from java.io import FileInputStream
+from java.io import FileOutputStream
+from java.io import IOException
+from java.lang import IllegalArgumentException
+from javax.xml.parsers import ParserConfigurationException
+
+from oracle.weblogic.deploy.util import FileUtils
+from org.xml.sax import SAXException
+
+from wlsdeploy.aliases import alias_utils
+from wlsdeploy.aliases.location_context import LocationContext
+from wlsdeploy.aliases.model_constants import ODL_CONFIGURATION
+from wlsdeploy.aliases.model_constants import MODEL_LIST_DELIMITER
+from wlsdeploy.aliases.wlst_modes import WlstModes
+from wlsdeploy.exception.expection_types import ExceptionType
+from wlsdeploy.logging.platform_logger import PlatformLogger
+from wlsdeploy.tool.util.alias_helper import AliasHelper
+from wlsdeploy.util import dictionary_utils
+
+from oracle.core.ojdl.weblogic.ODLConfiguration import CONFIG_DIR
+from oracle.core.ojdl.weblogic.ODLConfiguration import CONFIG_FILE
+from oracle.core.ojdl.logging.config import LoggingConfigurationDocument
+
+_ADD_JVM_NUMBER = "AddJvmNumber"
+_CLASS = "Class"
+_ENCODING = "Encoding"
+_ERROR_MANAGER = "ErrorManager"
+_FILTER = "Filter"
+_FORMATTER = "Formatter"
+_HANDLER = "Handler"
+_HANDLER_DEFAULTS = "HandlerDefaults"
+_HANDLERS = "Handlers"
+_LEVEL = "Level"
+_LOGGER = "Logger"
+_PROPERTIES = "Properties"
+_SERVERS = "Servers"
+_USE_PARENT_HANDLERS = "UseParentHandlers"
+
+
+class OdlDeployer(object):
+    """
+    Handle the ODL validation and configuration.
+    """
+    __class_name = 'OdlHelper'
+
+    def __init__(self, model, model_context, aliases, wlst_mode):
+        self.model = model
+        self.model_context = model_context
+
+        self.logger = PlatformLogger('wlsdeploy.deploy')
+        self.alias_helper = AliasHelper(aliases, self.logger, ExceptionType.DEPLOY)
+        self.wlst_mode = wlst_mode
+
+    def configure_odl(self, parent_dict, parent_location):
+        """
+        Apply the ODL configuration section of the model.
+        """
+        _method_name = 'configure_odl'
+        self.logger.entering(class_name=self.__class_name, method_name=_method_name)
+
+        odl_info = dictionary_utils.get_dictionary_element(parent_dict, ODL_CONFIGURATION)
+        if len(odl_info):
+            if not self.model_context.get_domain_typedef().is_jrf_domain_type():
```

Comment:
ODL is also installed into RestrictedJRF.

Reply:
This has been corrected and tested with Restricted JRF.